src/main/xar-resources/data/urlrewrite/urlrewrite.xml

```diff
@@ -280,191 +334,201 @@
         </tbody>
       </tgroup>
     </table>
-    <para>You do not need to explicitly declare the variables or the namespace. However doing so is
-      considered best practice, and you can add an external declaration for these variables at the
-      top of your XQuery. For instance:</para>
-    <programlisting language="xquery">declare namespace exist = "http://exist.sourceforge.net/NS/exist";
+    <para>These variables are pre-bound for simplicity and convenience. You do not need to
+      explicitly declare the variables or the <literal>exist</literal> namespace. However, doing so
+      is considered best practice, by adding this block to the prolog of your
+        <literal>controller.xq</literal> script. For instance:</para>
+    <programlisting language="xquery">declare namespace exist="http://exist.sourceforge.net/NS/exist";
 
+declare variable $exist:root external;
+declare variable $exist:prefix external;
+declare variable $exist:controller external;
 declare variable $exist:path external;
 declare variable $exist:resource external;
-declare variable $exist:controller external;
-declare variable $exist:prefix external;
-declare variable $exist:root external;</programlisting>
+</programlisting>
+    <para>These variables can also be accessed by any query within the controller hierarchy via the
+        <literal>request:get-attribute()</literal> function, e.g.,
+        <literal>request:get-attribute("$exist:prefix")</literal>.</para>
+    <para>Some further remarks about each variable:</para>
     <variablelist>
       <varlistentry>
         <term>
-          <code>exist:path</code>
+          <code>exist:root</code>
         </term>
         <listitem>
-          <para>Is bound to the last part of the request URL, i.e. after the section leading to the
-            collection containing the controller.xq. </para>
-          <para>For instance, if the resource <literal>example.xml</literal> resides within the same
-            collection as the controller query, the <literal>$exist:path</literal> variable would be
-            bound to the value <literal>/example.xml</literal>.</para>
+          <para>Is bound to the root of the current controller hierarchy. This may either point to
+            the file system or to a collection in the database. You can use this variable to locate
+            resources relative to the root of the application.</para>
+          <para>For example, assume that you want to process a request using stylesheet
+              <literal>db2xhtml.xsl</literal>, which could <emphasis>either</emphasis> be stored in
+            the <literal>/stylesheets</literal> directory in the root of the webapp or, if the app
+            is running from within the database, the corresponding <literal>/stylesheets</literal>
+            collection. You want your app to be able to run from either location. The solution is to
+            incorporate the value of the <literal>exist:root</literal> variable into your
```

Comment:
Shouldn't it be `<literal>$exist:root</literal>` here with a dollar sign to mark a variable?
This question applies to all the variables explained below.

Reply:
Good catch. The other variables already had the dollar sign, but it was missing in this case.

Reply:
Oops, the other variables marked with `<literal>` did, but not the ones marked with `<code>`.